02-microarray/pathway-analysis_microarray_03_gsea.Rmd

```diff
@@ -0,0 +1,493 @@
+  dplyr::pull(entrez_id)
+```
+
+Now let's take a look at the rows associated with one of the duplicated Entrez IDs.
+
+```{r}
+dge_mapped_df %>%
+  dplyr::filter(entrez_id == "336702")
+```
+
+We can see that the associated values vary for each row although the Entrez ID is the same.
+
+We do not want duplicated gene identifiers for the GSEA steps later, so let's keep the Entrez IDs associated with the higher t-statistic value.
+
+```{r}
+filtered_dge_mapped_df <- dge_mapped_df %>%
+  # Sort so that highest t-statistic values are at the top
+  dplyr::arrange(dplyr::desc(t)) %>%
+  # Filter out the duplicated rows using `dplyr::distinct()`-- this will keep
+  # the first row with the duplicated value thus keeping the row with the
+  # highest t-statistic value
+  dplyr::distinct(entrez_id, .keep_all = TRUE)
+```
+
+Let's check to see that we removed the duplicate Entrez IDs and kept the row with the higher t-statistic value.
+
+```{r}
+filtered_dge_mapped_df %>%
+  dplyr::filter(entrez_id == "336702")
+```
+
+Looks like we were able to successfully get rid of the duplicate gene identifiers and keep the observation with the higher t-statistic value!
+Note however, that a caveat in using this approach is that the genes that have duplicate identifiers could be enriched in a particular pathway/gene set and we may get an overly optimistic view of how perturbed that pathway truly is.
+
+**START REVIEW HERE**
+
+## Perform gene set enrichment analysis (GSEA)
+
+The goal of GSEA is to determine whether the members of a set of genes are randomly distributed throughout the pre-ranked gene list or primarily found at the top or bottom of the list.
+Genes are ranked based on phenotype so we expect that sets related to a distinct phenotype will tend to show the latter distribution [@clusterProfiler-book].
+
+### Determine our pre-ranked genes list
+
+The `GSEA()` function takes a pre-ranked (sorted) named vector of statistics, where the names in the vector are gene identifiers.
+This is step 1 -- gene-level statistics.
+
+```{r}
+# Let's create a named vector ranked based on the t-statistic values -- we
+# already sorted in the t-stastic values in descending order in the previous
+# chunk so we won't need to do that here
+t_vector <- filtered_dge_mapped_df$t
+names(t_vector) <- filtered_dge_mapped_df$entrez_id
+```
+
+Let's preview our pre-ranked named vector.
+
+```{r}
+# Look at first entries of the ranked t-statistic vector
+head(t_vector)
+```
+
+### Run GSEA using the `GSEA()` function
+
+The enrichment score (ES) for a pathway, a pathway-level statistic, is calculated using our gene-level statistics.
+Genes are ranked from most highly positive to most highly negative and weighted according to their gene-level statistic.
+A running score is calculated by starting with the most highly ranked genes and increasing the score when a gene is in the pathway and decreasing the score when a gene is not in the pathway.
+The ES is the maximum deviation from zero.
```

Comment:
Are you saying that ESs can be negative or positive where 0 indicates randomness? 

I think you could move this sentence up above the 355 line. Tell us intuitively what the score means and then if needed, we can get into how that's calculated.